gigascience/gigadb-website
# A11y - Improve a11y of help page

# Pull request for issue: #1378

This is a pull request for the following functionalities:

Fix a11y issues and improve a11y in help page:

* All round changes to help page:
  * ordered heading levels
  * content divided in sections
  * update content with more descriptive elements: `<dl>`, `<ol>`, `<code>`
  * add labels to datatype icons
  * remove inline styles
  * Autoformatted code
  * Descriptive labels in links to excel and pdf files
  * Replaced some uses of `<em>` element by styled italics as I considered `<em>` was being used only for styling, (e.g. `<em>GigaDB</em>` by `<span class="text-italic">GigaDB</span>"`
* Replaced inline style by utility class inline-block in other pages where inline styles were used

## How to test?

* Browse page http://gigadb.gigasciencejournal.com:9170/site/help and cross check that it looks ok visually
* Use a tool like AXE chrome extension to check accessibility issues in the main page content (header and footer issues out of scope)
* Extensive changes to the help page, potential checks:
  * Compare old and new text versions to confirm that text content is identical, e.g. using a tool such as this https://text-compare.com/ (I already verified old vs new content)

NOTE: if you don't see updated styles you might need to hard refresh browser and / or disable cache

## How have functionalities been implemented?

* When replacing previous markup with `<dl>` and `<ol>`, added new classes in less/current/pages/help.less to keep visual style
* Added a few utility classes in less/current/base/spacing.less and less/current/base/display.less to get rid of inline styles

## Any issues with implementation?

None

## Any changes to automated tests?

None

## Any changes to documentation?

None

## Any technical debt repayment?

None

## Any improvements to CI/CD pipeline?

None
